### ISSUE TITLE
After ugrading to OpenShift Logging 5.7.0 Operator CreateContainerConfigError

### ISSUE THREAD

**Issue by megian** [appuio/component-openshift4-logging#107]
After upgrading the OpenShift Logging 5.7.0 Operator stuck on start with the error
```
Error: container has runAsNonRoot and image will run as root (pod: "cluster-logging-operator-6b5d9c7495-8rhqh_openshift-logging(ea92698c-ce34-48b5-b458-47aba00c469d)", container: cluster-logging-operator)`.
```

On the openshift-logging namespace the pod security is set to `privileged`:
```
$ oc get ns openshift-logging -o yaml
apiVersion: v1
kind: Namespace
metadata:
  labels:
    ...
    pod-security.kubernetes.io/audit: privileged
    pod-security.kubernetes.io/enforce: privileged
    pod-security.kubernetes.io/warn: privileged
```

The OpenShift Logging Operator enforces the pod starting non-root:
```
spec:
  template:
    spec:
      containers:
      - name: cluster-logging-operator
        ...
        securityContext:
          allowPrivilegeEscalation: false
      securityContext:
        runAsNonRoot: true
```

The pod is actually started with the SCC `privileged-higher-prio`:
```
oc get po cluster-logging-operator-674c877f5b-w25p9 -n openshift-logging -o yaml | grep "openshift.io/scc"
openshift.io/scc: privileged-higher-prio
```

Either the namespace configuration is wrong or the operator deployment on upgrades.

## Steps to Reproduce the Problem

  1. Upgrade an existing OpenShift Logging from 5.6 to 5.7

## Actual Behavior
The Operator does not start. After removing the `securityContext.allowPrivilegeEscalation` and `securityContext. runAsNonRoot` the operator does startwithout issues.

## Expected Behavior
The upgrade does run without the operator stuck in root / non-root conflicts.

**Comment by megian** [comment]
Pre upgrade:
``
$ oc -n openshift-logging get csv
NAME                            DISPLAY                                          VERSION    REPLACES                        PHASE
cluster-logging.v5.6.6          Red Hat OpenShift Logging                        5.6.6      cluster-logging.v5.6.5          Succeeded
elasticsearch-operator.v5.7.1   OpenShift Elasticsearch Operator                 5.7.1      elasticsearch-operator.v5.7.0   Succeeded
jaeger-operator.v1.42.0-5       Red Hat OpenShift distributed tracing platform   1.42.0-5   jaeger-operator.v1.39.0-3       Succeeded
kiali-operator.v1.57.7          Kiali Operator                                   1.57.7     kiali-operator.v1.57.6          Succeeded
loki-operator.v5.6.6            Loki Operator                                    5.6.6      loki-operator.v5.6.5            Succeeded
servicemeshoperator.v2.3.3      Red Hat OpenShift Service Mesh                   2.3.3-0    servicemeshoperator.v2.3.2      Succeeded
``

**Comment by simu** [comment]
This SCC was created in https://github.com/appuio/component-openshift4-config/commit/e9106291008cd13e3eb6e3c6a79e1e712ccf8f09 to fix cluster upgrades to OCP 4.10 (some upgrade job pod failed without the patched privileged SCC, IIRC).

I've removed the SCC, and the pod now picks another non-standard SCC (`cert-exporter-scc`), and still fails to start with the same error message. I've found a RedHat solution article (https://access.redhat.com/solutions/7012133), which suggests that the issue is that an SCC with a higher priority than the default SCCs is picked (this would explain why `privileged-higher-prio` is picked) but the `cert-exporter-scc` doesn't set a priority at all, and therefore has the same priority as the builtin SCCs.

From what I can see, the actual root cause for the issue is that the cluster-logging-operator service account has permissions to use all SCCs present in the cluster, cf. `kubectl get clusterrole -l olm.owner=cluster-logging.v5.7.1 -oyaml`:

```
  - apiGroups:
    - security.openshift.io
    resources:
    - securitycontextconstraints
    verbs:
    - '*'
```

**Comment by simu** [comment]
Looks like granting verbs `get`, `list`, and `watch` on SCCs to the cluster-loggingo-operator is sufficient for the pod to start without errors and without picking a wrong SCC (not sure why the operator watches SCCs in the first place). If necessary, we could additionally grant all other verbs (except `use`) if the operator needs to be able to manage a custom SCC for the logging stack.

We'll have to figure out how we can cleanly patch the clusterrole since it's created/managed by OLM (I just edited it by hand to test my assumption).

**Comment by DebakelOrakel** [comment]
Seems fixed with 5.7.2

**Comment by megian** [comment]
> Seems fixed with 5.7.2

https://access.redhat.com/solutions/7012133 second that.
